icepyx/core/harmony.py

```diff
@@ -0,0 +1,334 @@
+from concurrent.futures import as_completed
+import datetime as dt
+import json
+from pathlib import Path
+import time
+from typing import Any, Dict, TypedDict, Union
+
+from _icepyx_version import version as ipx_version
+import harmony
+import requests
+
+from icepyx.core.auth import EarthdataAuthMixin
+
+# Sometimes harmony has problems (e.g., 500 bad gateway) and we need to retry.
+# 5 seconds seems like enough, but not always.
+REQUEST_RETRY_INTERVAL_SECONDS: int = 5
+
+
+class HarmonyTemporal(TypedDict):
+    # TODO: these are optional. Harmony can take a start without a stop or a
+    # stop without a start.
+    start: dt.datetime
+    stop: dt.datetime
+
+
+class HarmonyApi(EarthdataAuthMixin):
+    """
+    A client for interacting with the NASA Harmony API.
+
+    Attributes
+    ----------
+    harmony_client : harmony.Client
+        The Harmony API client.
+    job_ids : list of str
+        List of job IDs that have been placed with the Harmony API.
+
+    Methods
+    -------
+    get_capabilities(concept_id)
+        Retrieves the capabilities of a given dataset.
+    check_order_status(job_id)
+        Checks the status of a submitted Harmony job.
+    place_order(...)
+        Submits an order to Harmony and waits for completion.
+    download_granules(download_dir, overwrite)
+        Downloads all granules associated with past orders.
+    """
+
+    def __init__(self):
+        # initialize authentication properties
+        self.ipx_version = ipx_version
```

Comment:
Prefer to prefix this with an underscore, so that it's treated as a non-public API implementation detail.
```suggestion
        self._ipx_version = _ipx_version
```